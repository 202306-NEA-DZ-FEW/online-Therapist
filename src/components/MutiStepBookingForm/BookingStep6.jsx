import React from "react";
import Button from "../elements/Button";
import { useTranslation } from "next-i18next";

const BookingStep6 = ({
    formData,
    handleChangeInput,
    handlePrevStep,
    handleNextStep,
}) => {
    const { t } = useTranslation("booking");

    return (
        <div>
            <h3 className='mb-2 lg:text-2xl text-lg font-young text-DarkTeal '>
                {t("step6.header")}
            </h3>
            <p className='lg:text-lg font-atkinson text-gray-900'>
                {t("step6.paragraph1")}
                <br />
                {t("step6.paragraph2")}
            </p>

            <label htmlFor='specification' class='block mb-2 lg:pt-10 pt-6'>
                <textarea
                    id='specification'
                    value={formData.specification}
                    onChange={(e) => handleChangeInput(e)}
                    name='specification'
                    type='text'
                    rows='10'
                    class='block p-2.5 w-full lg:text-lg rtl:text-reverse font-atkinson text-gray-900 bg-gray-50 rounded-lg border hover:border-Teal border-gray-300 focus:ring-4 focus:ring-Teal'
                    placeholder={t("step6.placeholder")}
                ></textarea>
            </label>

            <div className='my-2 flex justify-between items-center lg:pt-6 pt-3'>
                <button onClick={handlePrevStep}>
                    <Button buttonText={t("step2.prev")} buttonSize='md' />
                </button>
                <button onClick={handleNextStep}>
                    <Button buttonText={t("step2.next")} buttonSize='sm' />
                </button>
            </div>
        </div>
    );
};

export default BookingStep6;
