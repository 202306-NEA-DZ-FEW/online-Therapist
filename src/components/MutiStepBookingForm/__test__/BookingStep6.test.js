import renderer from "react-test-renderer";
import BookingStep6 from "../BookingStep6";

it("renders correctly", () => {
    // Mock functions
    const mockHandleChangeInput = jest.fn();
    const mockHandlePrevStep = jest.fn();
    const mockHandleNextStep = jest.fn();

    // Mock initial formData
    const formData = {
        specification: "Sample specification", // Initialize as needed
        // Add other properties required by the component
    };

    const tree = renderer
        .create(
            <BookingStep6
                formData={formData}
                handleChangeInput={mockHandleChangeInput}
                handlePrevStep={mockHandlePrevStep}
                handleNextStep={mockHandleNextStep}
            />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});