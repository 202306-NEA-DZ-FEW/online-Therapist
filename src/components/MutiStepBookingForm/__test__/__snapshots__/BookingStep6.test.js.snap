// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3
    className="mb-2 lg:text-2xl text-lg font-young text-DarkTeal "
  >
    step6.header
  </h3>
  <p
    className="lg:text-lg font-atkinson text-gray-900"
  >
    step6.paragraph1
    <br />
    step6.paragraph2
  </p>
  <label
    class="block mb-2 lg:pt-10 pt-6"
    htmlFor="specification"
  >
    <textarea
      class="block p-2.5 w-full lg:text-lg rtl:text-reverse font-atkinson text-gray-900 bg-gray-50 rounded-lg border hover:border-Teal border-gray-300 focus:ring-4 focus:ring-Teal"
      id="specification"
      name="specification"
      onChange={[Function]}
      placeholder="step6.placeholder"
      rows="10"
      type="text"
      value="Sample specification"
    />
  </label>
  <div
    className="my-2 flex justify-between items-center lg:pt-6 pt-3"
  >
    <button
      onClick={[MockFunction]}
    >
      <div
        className="relative inline-flex items-center justify-center px-6 py-3 overflow-hidden font-medium transition duration-300 ease-out border-2 border-Teal rounded-lg shadow-md group text-xl w-32 h-14 bg-white hover:cursor-pointer"
      >
        <span
          className="absolute inset-0 flex items-center justify-center w-full h-full text-white duration-300 -translate-x-full bg-Teal group-hover:translate-x-0 ease"
        >
          <svg
            className="w-6 h-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 5l7 7m0 0l-7 7m7-7H3"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </span>
        <span
          className="absolute flex items-center justify-center w-full h-full text-Teal transition-all duration-300 transform group-hover:translate-x-full ease"
        >
          step2.prev
        </span>
        <span
          className="relative invisible"
        >
          $
          step2.prev
        </span>
      </div>
    </button>
    <button
      onClick={[MockFunction]}
    >
      <div
        className="relative inline-flex items-center justify-center px-6 py-3 overflow-hidden font-medium transition duration-300 ease-out border-2 border-Teal rounded-lg shadow-md group text-xl w-20 h-14 bg-white hover:cursor-pointer"
      >
        <span
          className="absolute inset-0 flex items-center justify-center w-full h-full text-white duration-300 -translate-x-full bg-Teal group-hover:translate-x-0 ease"
        >
          <svg
            className="w-6 h-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 5l7 7m0 0l-7 7m7-7H3"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </span>
        <span
          className="absolute flex items-center justify-center w-full h-full text-Teal transition-all duration-300 transform group-hover:translate-x-full ease"
        >
          step2.next
        </span>
        <span
          className="relative invisible"
        >
          $
          step2.next
        </span>
      </div>
    </button>
  </div>
</div>
`;
